hibernate-core/src/test/java/org/hibernate/orm/test/any/annotations/AnyTest.java

```diff
@@ -92,14 +117,66 @@ public void createTestData(SessionFactoryScope scope) {
 	public void dropTestData(SessionFactoryScope scope) {
 		scope.inTransaction(
 				session -> {
-					session.createQuery( "delete StringProperty" ).executeUpdate();
-					session.createQuery( "delete IntegerProperty" ).executeUpdate();
-					session.createQuery( "delete LongProperty" ).executeUpdate();
-					session.createQuery( "delete CharProperty" ).executeUpdate();
-
-					session.createQuery( "delete PropertyList" ).executeUpdate();
-					session.createQuery( "delete PropertyMap" ).executeUpdate();
-					session.createQuery( "delete PropertySet" ).executeUpdate();
+					session.createMutationQuery( "delete StringProperty" ).executeUpdate();
+					session.createMutationQuery( "delete IntegerProperty" ).executeUpdate();
+					session.createMutationQuery( "delete LongProperty" ).executeUpdate();
+					session.createMutationQuery( "delete CharProperty" ).executeUpdate();
+
+					session.createMutationQuery( "delete PropertyHolder" ).executeUpdate();
+					session.createMutationQuery( "delete PropertyList" ).executeUpdate();
+					session.createMutationQuery( "delete PropertyMap" ).executeUpdate();
+					session.createMutationQuery( "delete PropertySet" ).executeUpdate();
+					session.createMutationQuery( "delete LazyPropertySet" ).executeUpdate();
+				}
+		);
+	}
+
+	@Test
+	@TestForIssue( jiraKey = "HHH-15323")
+	public void testHqlCollectionTypeQuery(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					List<PropertySet> propertySets = session.createQuery(
+									"select p from PropertySet p where type(p.generalProperties) = IntegerProperty ",
+									PropertySet.class ).list();
+					assertEquals( 1, propertySets.size() );
+
+					PropertySet propertySet = propertySets.get( 0 );
+					assertEquals( 1, propertySet.getGeneralProperties().size() );
+
+					assertEquals( "age", propertySet.getGeneralProperties().get( 0 ).getName() );
+
+					propertySets = session.createQuery(
+									"select p from PropertySet p where type(p.generalProperties) = StringProperty ",
+									PropertySet.class ).list();
+					assertEquals( 1, propertySets.size() );
+
+					propertySet = propertySets.get( 0 );
+					assertEquals( 1, propertySet.getGeneralProperties().size() );
+
+					assertEquals( "name", propertySet.getGeneralProperties().get( 0 ).getName() );
+				}
+		);
+	}
+
+	@Test
+	@TestForIssue( jiraKey = "HHH-15323")
+	public void testHqlTypeQuery(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					List<PropertyHolder> propertyHolders = session.createQuery(
+									"select p from PropertyHolder p where type(p.property) = IntegerProperty ",
```

Comment:
To cover the setParameter issues I mentioned in a later comment in HHH-15323, could this test be iterated to use setParameter instead of a query literal?